client/my-sites/post-selector/selector.jsx

```diff
@@ -85,24 +89,31 @@ const PostSelectorPosts = React.createClass( {
 		}, SEARCH_DEBOUNCE_TIME_MS );
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( this.props.query, nextProps.query ) ) {
+			this.setState( {
+				requestedPages: [ 1 ]
```

Comment:
>does `searchTerm` also need to be reset here?

`search` is a prop passed in from the parent component and, for that matter, could be the reason that `query` changed.